solidity/solc-update-0.8.15-to-0.8.25.md

```diff
@@ -0,0 +1,98 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Solidity Version Upgrade Proposal](#solidity-version-upgrade-proposal)
+  - [Benefits to the OP Stack](#benefits-to-the-op-stack)
+      - [DevX](#devx)
+      - [Gas Optimization](#gas-optimization)
+      - [Features](#features)
+  - [Notable Features](#notable-features)
+  - [Notable Bug Fixes](#notable-bug-fixes)
+  - [Changelog Links](#changelog-links)
+  - [Action Items](#action-items)
+  - [Additional Notes](#additional-notes)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+|                     |                |
+| ------------------- | -------------- |
+| Author              | Michael Amadi  |
+| Created at          | 2024-10-02     |
+| Needs Approval From | Kelvin Fichter |
+| Other Reviewers     | Matt Solomon   |
+| Status              | In Review      |
+
+# Solidity Version Upgrade Proposal
+
+- **Current Version:** 0.8.15, 0.8.19, 0.8.25.
+- **Proposed Version:** 0.8.25.
+
+## Benefits to the OP Stack
+
+Explain specific feature(s) in the new version that would benefit the OP Stack codebase, with a
+clear explanation of their potential impact and usefulness.
+
+#### DevX
+
+- Allow named parameters in mapping types (>=0.8.18): Limits error when reading mappings of high dimensions when called using named parameters.
+- Allow defining custom operators for user-defined value types via using {f as +} for T global syntax (>=0.8.19): This greatly improves type safety.
+- Add support for NatSpec documentation in enum and struct definitions. (>= 0.8.20): Improves documentation and developer experience
+- Include NatSpec from events that are emitted by a contract but defined outside of it in userdoc and devdoc output (>= 0.8.20)
+- Allow qualified access to events from other contracts. (>= 0.8.21)
+
+#### Gas Optimization
+
+- Unchecked loop increments (>= 0.8.22)
+- Use `mcopy` instead of `mload`/`mstore` loop when copying byte arrays (>= 0.8.25)
```

Comment:
mcopy will also simplify our proofs and may improve performance there as a result

Reply:
Not sure if you suggest I add this or just a comment...